addons/web/static/src/env.js

```diff
@@ -74,6 +75,10 @@ export async function startServices(env) {
             const namedService = Object.assign(Object.create(service), { name });
             toStart.add(namedService);
         } else {
+            if (isStartingService) {
+                return;
+            }
+            isStartingService = true;
```

Comment:
This check should be inside of _startServices so that if a new service is added while the first start is still ongoing, it isn't called twice. The flag should also be reset to false after the services are started so that if a new service is added at a later point (eg by lazy-loading a bundle), that service can be started.